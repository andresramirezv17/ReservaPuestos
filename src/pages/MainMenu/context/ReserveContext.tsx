import React, { createContext, useEffect, useState } from 'react';
import moment from 'moment';
import { SearchParameters } from './models/SearchParameters';
import { WorkspacesSearched } from './models/WorkspacesSearched';
import { Floors } from './models/Floors';
import { Sections } from './models/Sections';
import * as ReserveServices from '../services/WorkspaceServices';

export interface WorkSpaceParameter {
  workspacesSearched?: WorkspacesSearched;
  floors?: Floors;
  sections?: Sections;
}

export interface propsSection {
  idFloor: number;
  idBuilding: number;
  name: string;
}

export const useStateContainer = (initialState: WorkSpaceParameter) => {
  const [workplaces, setworkplaces] = useState(
    initialState.workspacesSearched || { workplaces: [] },
  );

  const [floors, setfloors] = useState(
    initialState.floors || { floorsResult: [] },
  );
  const [sections, setsections] = useState(
    initialState.sections || { sectionsResult: [] },
  );

  const [parameters, setparameters] = useState('');
  const [building, setbuilding] = useState('Torre Central 2');
  const [floor, setfloor] = useState('Piso 1');
  const [section, setsection] = useState('Seccion A');
  const [date, setdate] = useState(
    moment(new Date()).add(1, 'days').format('DD/MM/YYYY'),
  );

  const searchWorkplaces = () => {
    ReserveServices.getSearchedWorkplaces().then((resultSearch) => {
      setworkplaces(resultSearch);
    });
  };
  const obtainFloors = () => {
    ReserveServices.getFloors().then((resultFloor) => setfloors(resultFloor));
  };

  const obtainSections = (id?: number) => {
    ReserveServices.getSections().then((resultSections) => {
      setsections(resultSections);
    });
  };

  const changeParameter = (parameter: string, value: string) => {
    if (parameter === 'piso') {
      setfloor(value);
    }
    if (parameter === 'section') {
      setsection(value);
    }
    if (parameter === 'date') {
      setdate(value);
    }
  };

  return {
    data: { workplaces, floors, sections, building, floor, section, date },
    mutations: {
      searchWorkplaces,
      obtainFloors,
      obtainSections,
      changeParameter,
    },
  };
};

export const ReserveContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

export interface SearchWorkspaceProvider {
  initialState?: WorkSpaceParameter;
}

export const ReserveProvider: React.FC<SearchWorkspaceProvider> = ({
  children,
  initialState = {},
}) => {
  const contextValue = useStateContainer(initialState);
  return (
    <ReserveContext.Provider value={contextValue}>
      {children}
    </ReserveContext.Provider>
  );
};
