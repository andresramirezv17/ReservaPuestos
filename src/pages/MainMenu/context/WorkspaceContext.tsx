import React, { createContext, useEffect, useState } from 'react';
import { UserWorkspace } from './models/UserWorkspace';
import * as WorkspaceService from '../services/WorkspaceServices';

export interface WorkSpaceState {
  workSpace?: UserWorkspace;
  varpueba?: boolean;
}

export const useStateContainer = (initialState: WorkSpaceState) => {
  const [workspaceUser, setworkspaceUser] = useState(
    initialState.workSpace || {},
  );
  const [prueba, setprueba] = useState(initialState.varpueba || true);

  useEffect(() => {
    WorkspaceService.getReserves(1).then((reserves) =>
      setworkspaceUser(reserves),
    );
  }, []);

  return { workspaceUser, prueba };
};

export const WorkspaceContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

export interface WorkspaceProvider {
  initialState?: WorkSpaceState;
}

export const UserWorkspaceProvider: React.FC<WorkspaceProvider> = ({
  children,
  initialState = {},
}) => {
  const contextValue = useStateContainer(initialState);
  return (
    <WorkspaceContext.Provider value={contextValue}>
      {children}
    </WorkspaceContext.Provider>
  );
};
