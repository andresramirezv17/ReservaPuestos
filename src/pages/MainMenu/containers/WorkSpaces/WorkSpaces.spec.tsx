import React from 'react';
import { WorkSpaces } from '.';
import { UserWorkspaceProvider } from '../../context/WorkspaceContext';
import { render, fireEvent } from '@testing-library/react';

describe('Workspace User test', () => {
  it('should match with snapshot', () => {
    const { container } = render(
      <UserWorkspaceProvider>
        <WorkSpaces />
      </UserWorkspaceProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
