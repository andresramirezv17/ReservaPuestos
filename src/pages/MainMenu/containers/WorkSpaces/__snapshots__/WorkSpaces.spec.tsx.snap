// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace User test should match with snapshot 1`] = `
<div>
  <div>
    <h5
      class="MuiTypography-root makeStyles-title-1 MuiTypography-h5"
    >
      Puestos reservados
    </h5>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Estos son los espacios que has reservado
    </p>
    <div
      style="width: 100%; display: inline-flex; flex-wrap: wrap; gap: 15px;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      />
    </div>
  </div>
</div>
`;
